### PR TITLE
fix: new execution of enricher

### PR DESCRIPTION
**Pull request by omrozowicz-splunk**
# Description

This fix change the way enricher is being executed. There are two scenarios:
1. Any change was issued in IF-MIB oid family -> in this case walk for IF-MIBs is being executed, and the data is enriched the same way as it was working before
2. There is a change only outside IF-MIB -> in this case only additional data was changed, so no need to trigger walk

Fixes # ADDON-43522

## Type of change

Please delete options that are not relevant.

- [x] Bug fix

## How Has This Been Tested?

I was testing many scenarios of deleting, adding, modifying and all of those at once on my local instance and I was checking mongodb contents every time. Also I installed this version on NOVA and modified enricher for a few times.

## Checklist

- [x] My commit message is [conventional](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] I have run pre-commit on all files before creating the PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: splunk_connect_for_snmp_poller/manager/poller.py [hunk 1]
```diff
@@ -131,8 +133,33 @@ def __check_inventory(self):
...
                         self.update_schedule_for_changed_conf(entry_key, ir, profiles)
 
+            if server_config_modified:
+                new_enricher = self._server_config.get("enricher", {})
+                if new_enricher != self._old_enricher:
+                    self.run_enricher_check(new_enricher, profiles, inventory_hosts)
             self.clean_job_inventory(inventory_entry_keys, inventory_hosts)
 
+    def run_enricher_check(self, new_enricher, profiles, inventory_hosts):
+        logger.debug(
```
Comment (weliasz) [inline splunk_connect_for_snmp_poller/manager/poller.py]: Maybe as this called on demand we should put info level? What do you think?

File: splunk_connect_for_snmp_poller/manager/poller_utilities.py [hunk 2]
```diff
@@ -274,6 +277,87 @@ def automatic_realtime_task(
...
+    splunk_indexes,
+):
+    ir = InventoryRecord(
+        host=inventory_host,
+        version="2c",
+        community="public",
+        profile=OidConstant.IF_MIB,
+        frequency_str="10",
+    )
+    ir.profile = OidConstant.IF_MIB
```
Comment (weliasz) [inline splunk_connect_for_snmp_poller/manager/poller_utilities.py]: What is the purpose of this assignment? ir.profile is already equal to OidConstat.IF_MIB

File: splunk_connect_for_snmp_poller/manager/poller_utilities.py [hunk 3]
```diff
@@ -274,6 +277,87 @@ def automatic_realtime_task(
...
+def _update_enricher_config_with_ifmib(
+    profiles,
+    inventory_host,
+    server_config,
+    splunk_indexes,
+):
+    ir = InventoryRecord(
+        host=inventory_host,
+        version="2c",
+        community="public",
```
Comment (weliasz) [inline splunk_connect_for_snmp_poller/manager/poller_utilities.py]: Maybe version and community should be taken from config?

File: splunk_connect_for_snmp_poller/manager/poller_utilities.py [hunk 4]
```diff
@@ -274,6 +277,87 @@ def automatic_realtime_task(
...
+    additional_enricher_varbinds = (
+        extract_network_interface_data_from_additional_config(server_config)
+    )
+    mongo.update_static_data_for_one(inventory_host, additional_enricher_varbinds)
+
+
+def is_ifmib_different(old_enricher, new_enricher):
+    new_if_mib = multi_key_lookup(new_enricher, ("oidFamily", "IF-MIB"))
+    old_if_mib = multi_key_lookup(old_enricher, ("oidFamily", "IF-MIB"))
+    return bool(new_if_mib != old_if_mib)
```
Comment (weliasz) [inline splunk_connect_for_snmp_poller/manager/poller_utilities.py]: Why is bool call required? != operator should return Boolean or am I wrong?

File: splunk_connect_for_snmp_poller/manager/poller_utilities.py [hunk 5]
```diff
@@ -274,6 +277,87 @@ def automatic_realtime_task(
...
+    mongo.delete_oidfamilies_from_static_data(inventory_host, families_to_delete)
+    additional_enricher_varbinds = (
+        extract_network_interface_data_from_additional_config(server_config)
+    )
+    mongo.update_static_data_for_one(inventory_host, additional_enricher_varbinds)
+
+
+def is_ifmib_different(old_enricher, new_enricher):
+    new_if_mib = multi_key_lookup(new_enricher, ("oidFamily", "IF-MIB"))
+    old_if_mib = multi_key_lookup(old_enricher, ("oidFamily", "IF-MIB"))
```
Comment (weliasz) [inline splunk_connect_for_snmp_poller/manager/poller_utilities.py]: How about using constants?

File: splunk_connect_for_snmp_poller/manager/poller.py (hunk 1 above)
Comment (omrozowicz-splunk) [reply splunk_connect_for_snmp_poller/manager/poller.py]: Good idea, changed

File: splunk_connect_for_snmp_poller/manager/poller_utilities.py (hunk 2 above)
Comment (omrozowicz-splunk) [reply splunk_connect_for_snmp_poller/manager/poller_utilities.py]: Right, I missed it

File: splunk_connect_for_snmp_poller/manager/poller_utilities.py (hunk 4 above)
Comment (omrozowicz-splunk) [reply splunk_connect_for_snmp_poller/manager/poller_utilities.py]: Right, not required, removed it

File: splunk_connect_for_snmp_poller/manager/poller_utilities.py (hunk 5 above)
Comment (omrozowicz-splunk) [reply splunk_connect_for_snmp_poller/manager/poller_utilities.py]: Updated

File: splunk_connect_for_snmp_poller/manager/poller_utilities.py (hunk 3 above)
Comment (omrozowicz-splunk) [reply splunk_connect_for_snmp_poller/manager/poller_utilities.py]: I am not sure about the situation when we can have many profiles for one host and they can have different communities and versions, in that situation which community and version should I choose?

**Review by weliasz — APPROVED** [review]

**Pull request merged**